### PR TITLE
Use Ansible encoder for dumping event data

### PR DESCRIPTION
**Pull request by AlanCoding**
Connect https://github.com/ansible/ansible-runner/issues/330

See: https://github.com/ansible/ansible/blob/devel/lib/ansible/plugins/callback/json.py#L38

I think we can import Ansible here???

```
ansible-runner --playbook test.yml run alan 
Vault password: 

PLAY [localhost] ***************************************************************

TASK [set_fact] ****************************************************************
ok: [localhost]

TASK [debug] *******************************************************************
ok: [localhost] => {
    "msg": "My password value is hell workld111"
}

PLAY RECAP *********************************************************************
localhost                  : ok=2    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   


```

### CODE REVIEW

File: ansible_runner/display_callback/events.py [hunk 1]
```diff
@@ -28,6 +28,9 @@
 import threading
 import uuid
 
+# Ansible
+from ansible.parsing.ajson import AnsibleJSONEncoder
```
Comment (matburt) [inline ansible_runner/display_callback/events.py]: This is heading into some dangerous territory as Runner doesn't import Ansible directly right now. How portable is this? Do we *need* to do this?

Comment (AlanCoding) [reply ansible_runner/display_callback/events.py]: This bug throws errors in the callback plugin, I would say it's a must-fix.

The alternative is to copy over @ryanpetrello's

https://github.com/ansible/awx/blob/devel/awx/main/queue.py#L23

We could switch AWX to import runner's version of the encoder, eventually. That could move us in the direction of making ansible-runner the de-facto python library for Ansible variable encoder / decoders. I don't necessarily think that's a bad idea either.

Keep in mind, AWX already has _other_ encoders and decoders for unsafe variables. We may have yet un-discovered bugs for other types.

Comment (matburt) [reply ansible_runner/display_callback/events.py]: I think I'd definitely like to pull in *that* old encoder if it gets us there. I'd like to keep from importing Ansible *at all* if there is literally any other way around an issue like this.

Comment (AlanCoding) [reply ansible_runner/display_callback/events.py]: here? https://github.com/ansible/ansible-runner/blob/master/ansible_runner/utils.py

Comment (AlanCoding) [reply ansible_runner/display_callback/events.py]: No, wait, I have even more questions. If I put this in utils, it still wouldn't be a simple import. In cases where these files import from other parts of ansible_runner, it doesn't import from the ansible_runner module.

https://github.com/ansible/ansible-runner/blob/master/ansible_runner/display_callback/module.py#L33

There's a lot of baggage with how the imports are done. I don't necessarily feel more confident importing from another file (wherever that may be) in ansible_runner. So I guess the option on the table is to just define the class right in that file.

Comment (matburt) [reply ansible_runner/display_callback/events.py]: If we think we might expand on this in the future, which seems likely, then I feel like we should possibly create a `ansible_runner/encoders.py`

**Comment by AlanCoding** [comment]
working with new version

```
ansible-runner --playbook test.yml run alan 
Vault password: 

PLAY [localhost] ***************************************************************

TASK [set_fact] ****************************************************************
ok: [localhost]

TASK [debug] *******************************************************************
ok: [localhost] => {
    "msg": "My password value is hell workld111"
}

PLAY RECAP *********************************************************************
localhost                  : ok=2    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   

```

**Comment by AlanCoding** [comment]
Handling of unsafe vars is left out of this, but I realize that may need addressing if

https://github.com/ansible/ansible/pull/60602

is merged.

**Pull request merged**